csharp/ql/src/semmle/code/csharp/PrintAst.qll

```diff
@@ -576,19 +690,21 @@ final class BaseTypesNode extends PrintAstNode, TBaseTypesNode {
 
   override Location getLocation() { none() }
 
-  override BaseTypeNode getChild(int childIndex) {
+  override TopLevelTypeMentionNode getChild(int childIndex) {
     childIndex = 0 and
-    result.getBaseType() = valueOrRefType.getBaseClass() and
-    result.getDerivedType() = valueOrRefType
+    result.getTypeMention().getType() = valueOrRefType.getBaseClass() and
+    result.getTypeMention().getTarget() = valueOrRefType
     or
-    result.getBaseType() =
-      rank[childIndex](ValueOrRefType base, string name |
-        base = valueOrRefType.getABaseInterface() and
-        name = base.toString()
+    result.getTypeMention() =
+      rank[childIndex](TypeMention t, Location l |
+        t =
+          any(TypeMention tm |
+            tm.getType() = valueOrRefType.getABaseInterface() and tm.getTarget() = valueOrRefType
+          ) and
```

Comment:
`t.getType() = valueOrRefType.getABaseInterface() and t.getTarget() = valueOrRefType`